clarity/src/vm/database/clarity_db.rs

```diff
@@ -763,7 +763,7 @@ impl<'a> ClarityDatabase<'a> {
             .ok();
         match x_opt.flatten() {
             None => Ok(None),
-            Some(x) => ContractAnalysis::deserialize(&x).map(|out| Some(out)),
+            Some(x) => ContractAnalysis::deserialize(&x).map(Some),
```

Comment:
Maybe outside the scope of this PR, but this `match` statement is confusing. The function should be rewritten as:

```rust
        self
            .store
            .get_metadata(contract_identifier, AnalysisDatabase::storage_key())
            // treat NoSuchContract error thrown by get_metadata as an Option::None --
            //    the analysis will propagate that as a CheckError anyways.
            .ok()
            .flatten()
            .map(|x| ContractAnalysis::deserialize(&x) )
            .transpose()
```

Reply:
I agree. Don't mind doing now.